packages/@uppy/companion/src/server/controllers/send-token.js

```diff
@@ -29,8 +30,8 @@ module.exports = function sendToken (req, res, next) {
     const allowedClients = req.uppy.options.clients
     // if no preset clients then allow any client
     if (!allowedClients || hasMatch(origin, allowedClients) || hasMatch(parseUrl(origin).host, allowedClients)) {
-      // @todo do a more secure client version check, see https://www.npmjs.com/package/semver
-      return res.send(clientVersion ? htmlContent(uppyAuthToken, origin) : oldHtmlContent(uppyAuthToken, origin))
+      const allowsStringMessage = versionCmp.gte(clientVersion, '1.0.2')
```

Comment:
```suggestion
      const allowsStringMessage = clientVersion ? versionCmp.gte(clientVersion, '1.0.2') : false
```

Reply:
This fails with Uppy Client pre this PR, since no version is sent. I’m not sure my fix is the most robust, we should not forget Companion can’t assume any version will be sent (until 2.0).

Maybe this check should be moved to the version check method, so we don’t have to remember it in every condition?

To test, open Uppy homepage locally with `npm run web:start` and try to connect to Instagram.

Reply:
> This fails with Uppy Client pre this PR, since no version is sent.

versions were sent before this PR and still are through the `uppyVersion` query param

Reply:
1.0.0 did not send the version right?

Reply:
yes, so 1.0.0 will receive the old version of token without JSON.stringify

Reply:
so `clientVersion` would be `null` here, could we default it to '1.0.0' or something if it is not available?

Reply:
there's a ternary there and we are only doing a version comparison when `clientVersion` is available. Meaning that we are not comparing when `clientVersion` is null

Reply:
```
::1 - - [03/Jun/2019:14:11:25 +0000] "GET /instagram/callback?access_token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&raw= HTTP/1.1" 302 1118 "http://localhost:4000/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:67.0) Gecko/20100101 Firefox/67.0"
 TypeError: semver.coerce is not a function
    at Object.exports.gte (/Users/artur/Projects/transloadit/uppy/packages/@uppy/companion/src/server/helpers/version.js:10:15)
    at sendToken (/Users/artur/Projects/transloadit/uppy/packages/@uppy/companion/src/server/controllers/send-token.js:33:46)
    at Layer.handle [as handle_request] (/Users/artur/Projects/transloadit/uppy/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/artur/Projects/transloadit/uppy/node_modules/express/lib/router/route.js:137:13)
    at exports.verifyToken (/Users/artur/Projects/transloadit/uppy/packages/@uppy/companion/src/server/middlewares.js:25:3)
    at Layer.handle [as handle_request] (/Users/artur/Projects/transloadit/uppy/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/artur/Projects/transloadit/uppy/node_modules/express/lib/router/route.js:137:13)
    at exports.hasSessionAndProvider (/Users/artur/Projects/transloadit/uppy/packages/@uppy/companion/src/server/middlewares.js:15:10)
    at Layer.handle [as handle_request] (/Users/artur/Projects/transloadit/uppy/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/artur/Projects/transloadit/uppy/node_modules/express/lib/router/route.js:137:13)
```

Reply:
err ... did this happen locally? What version of `semver` do you have installed? Did you install `semver` before this PR?

Reply:
It was an incorrect error indeed (fixed by bootstrap, sorry), but this issue remains, I’ve updated the error message above.

> there's a ternary there and we are only doing a version comparison when clientVersion is available

Where is the ternary? That’s the issue that I see, we are doing the check without checking checking if `clientVersion` is `null` first. Hence my suggested code with ternary.

Reply:
> Where is the ternary? 

In your code suggestion. Code review is very confusing these days, I thought that was my code 😂

Reply:
could we default the clientVersion to '1.0.0' where it's initially read from the request? then we don't have to do these ternaries every time we use it.

Reply:
> could we default the clientVersion to '1.0.0' where it's initially read from the request

done 👍